src/koa-npm-resolution.ts

```diff
@@ -0,0 +1,32 @@
+import * as Koa from 'koa';
+import {resolve as resolvePath} from 'path';
+import {parse as parseURL} from 'url';
+import esmSpecifierTransform from './koa-esm-specifier-transform';
+import resolveNPMSpecifier from './support/resolve-npm-specifier';
+
+export type Options = {
+  /* On-disk package root path used for NPM package resolution; defaults to
+     current folder. */
+  packageRoot?: string,
+  /* When serving the package from a base other than root.  If this option is
+     given, no specifier resolution will be attempted for documents served
+     outside the baseHref. */
+  baseHref?: string,
+};
+
+export default middleware;
+export function middleware(options: Options = {}): Koa.Middleware {
+  const onDiskPackageRoot = resolvePath(options.packageRoot || '.');
+  const baseHref =
+      (options.baseHref || '').replace(/^\//, '').replace(/[^\/]+$/, '');
+  return esmSpecifierTransform((baseURL: string, specifier: string) => {
+    const path = (parseURL(baseURL).pathname || '/').replace(/^\//, '');
+    if (!path.startsWith(baseHref)) {
```

Comment:
I think you really want path-specific logic here, for cases like base=/foo/bar, path=/foo/barbaz.

See https://www.npmjs.com/package/path-is-inside

Reply:
I renamed some stuff here, extracted a couple of helper functions but `path-is-inside` didn't work for me because my values were `foo/bar/` instead of `/foo/bar`.  I did this because a `<base href="foo/bar">` is effectively `<base href="foo/">` where relative paths are concerned and so I strip off `/[^\/]+$/` from the value to work with it.